### PR TITLE
refactor/dropdown menu

### LINKED ISSUE #15: Log out & changing network should be in a dropdown menu
This doesn't really look amazing yet
image

1. Let's have a cyan circle with a 👤 in it
2. On hover / click show the wallet address modal
image
3. Add three buttons there: logout, Polygon & Ethereum.

### PR DESCRIPTION
**Pull request by Syncretik**
resolves #15 

significant modal logic refactoring as we were mixing native + synthetic event handlers which blocked all click actions inside modal.

### CODE REVIEW

File: components/Transactions.tsx [hunk 1]
```diff
@@ -9,9 +9,11 @@ export default function Transactions() {
   const { chain } = useNetwork();
 
   useEffect(() => {
+    console.log({ chain });
```
Comment (gglucass) [inline components/Transactions.tsx:12]: let's remove these instances of console.log

Comment (gglucass) [reply components/Transactions.tsx:12]: a lol totally missed the draft 🤦

Comment (Riglanto) [reply components/Transactions.tsx]: I would avoid adding reviewers to the draft, unless you want comments :)

File: components/Header.tsx [hunk 2]
```diff
@@ -61,14 +54,6 @@ export default function Header({ address, isConnected }: Props) {
           Log in
         </button>
       )}
-      <button
```
Comment (Riglanto) [inline components/Header.tsx:64]: Are we not going to support chain switching anymore?

File: pages/_app.tsx [hunk 3]
```diff
@@ -70,17 +70,6 @@ export default function App({ Component, pageProps }: AppProps) {
     dialogRef.current?.close();
   };
 
-  useEffect(() => {
```
Comment (Riglanto) [inline pages/_app.tsx:73]: Have we decided to remove click outside to close function? Maybe let's create a ticket to add it back if we like it?

Comment (Syncretik) [reply pages/_app.tsx:73]: the previous implementation stopped all event propagation in the dialog box and I was trying to get the PR ready for release last wednesday. we can make a low-priority ticket to re-add #37

File: components/Header.tsx (hunk 2 above)
Comment (Syncretik) [reply components/Header.tsx:64]: we decided to remove for mvp, to add back later
Screen Shot 2023-05-30 at 11 12 05 PM

**Pull request merged**